packages/client/lib/sync/fetcher/blockfetcherbase.ts

```diff
@@ -115,7 +115,7 @@ export abstract class BlockFetcherBase<JobResult, StorageItem> extends Fetcher<
     // Some jobs could have refetch tasks enqueued, so it is better to find
     // the first and last by examining each job.
     while (this.in.length > 0) {
-      const job = this.in.remove()
+      const job = this.in.dequeue()
```

Comment:
Is this OK?

Reply:
Yes because i changed the Heap package we're using.

This solved one problem but created another, so this particular change is something I'm still tinkering with.